### PR TITLE
Improve user facing documentation and samples to be less QPC centric

### LINKED ISSUE #132: Improve user facing documentation and samples to be less QPC centric
## Feature description:
Improve user-facing documentation and samples to be less QPC centric

1. Break the meta-data and slices into their own apidoc files.  They will never be together so let's not show them together in swagger. Create a fake path so that the apidoc does not have validation errors in the swagger editor.
2. Create a sample.tar.gz that contains metadata and 2 slices (each with only 1 host)
3. Document in the README how the data should be formatted for Yupana.  This could follow the outline below:
    - Create a metadata section that describes the metadata format.  Point to the API doc for the meta-data.  Give an embedded example in JSON.
    - Create a slice section that describes a real slice.  Point to the API doc for the slice.  Give an embedded example in JSON.
    - Describe how the tar.gz is constructed (e.g. naming of files).  Point to the example tar.gz.

### PR DESCRIPTION
**Pull request by Dostonbek1**
- Added data description section to README;
- Separated metadata and reportslices into different files; 
- Created sample.tar.gz file

closes #132

### CODE REVIEW

File: docs/metadata.swagger.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  version: 1.0.0
+  title: Swagger Metadata
+  contact:
+    email: "quipucords@redhat.com"
+  license:
+    name: "GPL 3.0"
+    url: "https://www.gnu.org/licenses/gpl-3.0.txt"
+
+paths:
+  /hosts:
```
Comment (kholdaway) [inline docs/metadata.swagger.yml]: Let's make this path something like `/dummy` so it is clearly not a real API endpoint.  The description can also indicate that the endpoint does not exist.

File: docs/metadata.swagger.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+openapi: "3.0.0"
```
Comment (kholdaway) [inline docs/metadata.swagger.yml]: Just call this file metadata.yml

File: docs/reportslices.swagger.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+openapi: "3.0.0"
```
Comment (kholdaway) [inline docs/reportslices.swagger.yml]: Just call this file `report_slice.yml`

File: docs/reportslices.swagger.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  version: 1.0.0
+  title: Swagger Report Slices
+  contact:
+    email: "quipucords@redhat.com"
+  license:
+    name: "GPL 3.0"
+    url: "https://www.gnu.org/licenses/gpl-3.0.txt"
+
+paths:
+  /hosts:
```
Comment (kholdaway) [inline docs/reportslices.swagger.yml]: Same as above `/dummy`

File: docs/metadata.swagger.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        - hosts
+      summary: Read the entire list of hosts
+      description: Fake path created to avoid validation errors
+      responses:
+        '200':
+          description: Successfully read the hosts list.
+
+components:
+  schemas:
+    MetadataTarPath:
```
Comment (kholdaway) [inline docs/metadata.swagger.yml]: MetadataTarPath is no longer part of this model.  It was only used when we had a single JSON file with everything.

File: docs/reportslices.swagger.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        - hosts
+      summary: Read the entire list of hosts
+      description: Fake path created to avoid validation errors
+      responses:
+        '200':
+          description: Successfully read the hosts list.
+
+components:
+  schemas:
+    ReportSliceTarPath:
```
Comment (kholdaway) [inline docs/reportslices.swagger.yml]: This shouldn't be in this model anymore.  This was only used for the JSON when everything was combined.

File: README.rst [hunk 7]
```diff
@@ -132,8 +132,97 @@ To lint the code base ::
 
     tox -e lint
 
+Formatting Data for Yupana
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+Data sent to Yupana should include the following sections in given format (JSON):
```
Comment (kholdaway) [inline README.rst]: #### Let's give a structural overview to start.  For example, 

Data sent to Yupana should be a set of JSON files compressed in a `tar.gz file.` The tag.gz file should contain 1 metadata JSON file and 1 to many report slice JSON files.  Below we describe the metadata and report slice JSON.

File: README.rst [hunk 8]
```diff
@@ -132,8 +132,97 @@ To lint the code base ::
...
+    1. Metadata. Data should include metadata section which contains metadata information about the report and sources, like an example below: ::
```
Comment (kholdaway) [inline README.rst]: Metadata should include information about the sender of the data, Host Inventory API version, and the report slices included in the `tar.gz` file.  Below is a sample metadata section.

File: README.rst [hunk 9]
```diff
@@ -132,8 +132,97 @@ To lint the code base ::
...
+                "2dd60c11-ee5b-4ddc-8b75-d8d34de86a34": {
+                    "number_hosts": 1
+                },
+                "eb45725b-165a-44d9-ad28-c531e3a1d9ac": {
+                    "number_hosts": 1
+                }
+            }
+        }
+    
+       An API specification of the metadata can be found `here`_.
```
Comment (kholdaway) [inline README.rst]: Should point to a URL.  May be a chicken and egg problem.

File: README.rst [hunk 10]
```diff
@@ -132,8 +132,97 @@ To lint the code base ::
...
+
+    2. Report Slices. Report slices section of the data contains the general information about servers and hosts
```
Comment (kholdaway) [inline README.rst]: Report slices are a section of the host inventory data.  They are limited to a maximum size of 10K hosts.  Slices with more than 10K hosts will be discarded as a validation error.  Below is a sample report slice.
 
    
 
 
Leave a comment

Attach files by dragging & dropping, selecting or pasting them.
        generated during a scan. The report is separated into slices where the data is too large. Following is an example

File: README.rst [hunk 11]
```diff
@@ -132,8 +132,97 @@ To lint the code base ::
...
+
+    2. Report Slices. Report slices section of the data contains the general information about servers and hosts 
+       generated during a scan. The report is separated into slices where the data is too large. Following is an example 
+       of how each report slice should be formatted: ::
+
+        {
+            "report_slice_id": "2dd60c11-ee5b-4ddc-8b75-d8d34de86a34",
+            "hosts": [
+                {
+                    "display_name": "7cent.cfoo.example.com",
```
Comment (kholdaway) [inline README.rst]: Let's use an example that has canonical facts.

File: README.rst [hunk 12]
```diff
@@ -132,8 +132,97 @@ To lint the code base ::
...
+                        "number_of_sockets": 2,
+                        "cores_per_socket": 1
+                    }
+                }
+            ]
+        }
+       
+       An API specification of the report slices can be found `here.`_
+
+Tar.gz Format of the Data
```
Comment (kholdaway) [inline README.rst]: I think all of this information should be moved up to before the metadata and slice sections.  Let's describe the tar.gz structure before we dive into samples of the JSON.

File: docs/metadata.swagger.yml (hunk 5 above)
Comment (Dostonbek1) [reply docs/metadata.swagger.yml]: What should I use in place of it? Should I rename it to something else?

Comment (kholdaway) [reply docs/metadata.swagger.yml]: You renamed it.

**Comment by kholdaway** [comment]
Looks good to me but I want @abaiken to review it.

File: docs/report_slices.yml [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+      type: object
+      properties:
+        report_slice_id:
+          type: "string"
+          description: "The Insights report slice identifier"
+          example: "abc075ac-84d3-4b62-9f5c-a418a36f802d"
+        hosts:
+          type: "array"
+          items:
+            $ref: "https://raw.githubusercontent.com/RedHatInsights/insights-host-inventory/system_profile_changes/swagger/api.spec.yaml#components/schemas/CreateHostIn"
```
Comment (abaiken) [inline docs/report_slices.yml]: I still get validation errors in the swagger editor for some reason for both this and the `metadata.yml` (but maybe they are expected?) Also, this is kind of the same problem we had with the swagger in quipucords.  Linking to the host inventory api spec like this makes it seem like we are expecting satellite to send us their data with the `account` field already populated in each host. When in reality - for qpc at least - yupana is going to be populating the `account` field. I'm not sure how we want to handle this. @kholdaway what are your thoughts?

Comment (abaiken) [reply docs/report_slices.yml]: After talking to Kevan, we can ignore the validation errors. We can also leave the link to the host inventory api - but we should update the readme (probably around line 235) to say that Yupana will populate the `account` field for each host.

Comment (Dostonbek1) [reply docs/report_slices.yml]: @kholdaway What do you suggest we write about the `account` field in the README?

Comment (kholdaway) [reply docs/report_slices.yml]: see slack

**Pull request merged**